tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/domain/Filer.java

```diff
@@ -40,100 +39,79 @@ public Filer setLanguagePlugin(LanguagePlugin languagePlugin) {
     public FileVisitResult decideOnDirectory(Path directory) {
         return FileVisitResult.CONTINUE;
     }
-
-    public List<String> prepareFile(List<String> file) {
-        return file;
-    }
-
-    List<CommentStyleFileFilter> getCommentStyleFileFilters() {
-        return new ImmutableList.Builder<CommentStyleFileFilter>()
-                .add(
-                        // Basic java
-                        new CommentStyleFileFilter(
-                                // JavaishCommentSyntax
-                                CommentSyntax.newBuilder()
-                                        .addSingleLineComment("\\/\\/")
-                                        .addMultiLineComment("\\/\\*+", "\\*+\\/")
-                                        .build()))
-                .add(
-                        new CommentStyleFileFilter(
-                                // XML comment syntax
-                                CommentSyntax.newBuilder()
-                                        .addMultiLineComment("/\\*", "\\*/")
-                                        .build()))
-                //TODO: add rest of the comment syntaxes
-                .build();
+    
+    public void visitFileExceptionWrapper(Path source, Path relativePath) {
+        try {
+            visitFile(source, relativePath);
+        } catch (IOException ex) {
+            throw new RuntimeException();
+        }
     }
-
-    protected boolean skipFile(Path file) {
-        List<String> nameSkipList =
-                Arrays.asList(new String[] {".tmcrc", "metadata.yml"});
-        for (String item : nameSkipList) {
-            if (file.getFileName().toString().equals(item)) {
-                return true;
-            }
+    
+    private void visitFile(Path source, Path relativePath) throws IOException {
+        Path destination = toPath.resolve(relativePath);
+        if (skipFilename(source)) {
+            return;
         }
-
-        nameSkipList =
-                Arrays.asList(new String[] {"Hidden"});
-        for (String item : nameSkipList) {
-            if (file.getFileName().toString().contains(item)) {
-                return true;
-            }
+        if (looksLikeBinary(source)) {
+            justCopy(source, destination);
+        } else {
+            copyWithFilters(source, destination);
+        }
+    }
+    
+    private boolean skipFilename(Path source) {
+        logger.info("Looking into file: {} ", source);
+        String skipRegex = "\\.tmcrc|metadata\\.yml|(.*)Hidden(.*)";
+        if (source.getFileName().toString().matches(skipRegex)) {
+            logger.info("Skipping file: {} ", source);
+            return true;
         }
         return false;
```

Comment:
Now that looking for this method, let's change the log messages a bit
Let's have instead these two log messages
`logger.info("Skipping file: {} ", source);` <- remains unchanged 
`logger.info("Not skipping file: {} ", source);` <- just before the `return false`
and let's drop the first looking into file...